packages/tools/guiEditor/src/globalState.ts

```diff
@@ -163,10 +164,37 @@ export class GlobalState {
         this.onSelectionChangedObservable.notifyObservers();
     }
 
+    private _findParentControlInTexture(texture: AdvancedDynamicTexture, searchedControl: Control) {
+        const searchList = [texture.rootContainer];
+        while (searchList.length > 0) {
+            const current = searchList.splice(0,1)[0];
+            const children = current._children;
+            if (children.indexOf(searchedControl) !== -1) {
+                return current;
+            }
+            for (const child of children) {
+                if (child instanceof Container) {
+                    searchList.push(child);
+                }
+            }
+        }
+        return null;
+    }
+
     public deleteSelectedNodes() {
         for (const control of this.selectedControls) {
-            this.guiTexture.removeControl(control);
-            this.liveGuiTexture?.removeControl(control);
+            const guiTextureParent = this._findParentControlInTexture(this.guiTexture, control);
+            guiTextureParent?.removeControl(control);
+            if (this.liveGuiTexture) {
+                const allDescendants = control.getDescendants();
+                for (const descendant of allDescendants) {

```

Comment:
Is it necessary to remove all descendants? If we remove all direct children of the control, shouldn't that take all the descendants with them?

Reply:
Removing the direct descendants of a control does work on the Editor GUI Texture, but I think the Live GUI texture is organized in a different way, because I tried that and it didn't work :(

Reply:
Weird...so what happens to the descendants? Do they become children of the root?

Reply:
Yeah, they're children of the root on the Live texture

Reply:
Strange...we should definitely look into this more later on. Either way, this PR LGTM!